src/index.ts

```diff
@@ -366,8 +366,14 @@ export function serialize(
 /**
  * URL-decode string value. Optimized to skip native call when no %.
  */
-function decode(str: string): string {
-  return str.indexOf("%") !== -1 ? decodeURIComponent(str) : str;
+function decode(str: string): string | undefined {
+  if (str.indexOf("%") === -1) return str;
+
+  try {
+    return decodeURIComponent(str);
+  } catch (e) {
+    return undefined;
```

Comment:
Honestly I don't see a harm in returning the string here

It's just the value of a key

@climba03003 @uzlopak wdyt?

```suggestion
    return str;
```

Reply:
Returning a undecoded string can have implications downstream in user code, for example checking a user ID exists or parsing JSON. These should definitely fail with the original string.

Reply:
Okay, but does the HTTP State Management Mechanism (RFC 6265) require that the cookie values are properly URI-escaped?

I think you can in theory receive sth like `Cookie: name=%heyhey`, which fails URI decoding but should be valid -- URI decoding is just good practice when possible

It's this PR that's making the breaking change to return `undefined` if URI decoding fails, and I don't see the reason

It's been a while since I've read the RFC though, feel free to correct me if I'm wrong

Reply:
I think when you implement the `serialize` as using `encode`.
You MUST expect the cookies provided by `parse` can be decoded.

`serialize` and `parse` should considered as pair, if something comes in is not expected. It is high possibility not provided by this application.

Reply:
> I think when you implement the serialize as using encode -> You MUST expect the cookies provided by parse can be decoded.

And where in the RFC does it say that? You encode to protect the dev from breaking the header when they use things like `=`, special chars in the cookie header that must be escaped here

This isn't a signed cookie, the client is free to alter and send the value it wants

I'm highlighting again that currently it does return the string if decoding fails, and this PR changes the behavior to return `undefined`, and numerous cookies that are parsed by the cookie package, 51M downloads per week, will become invalid

Reply:
>> And where in the RFC does it say that?

> To maximize compatibility with user agents, servers that wish to store arbitrary data in a cookie-value SHOULD encode that data, for example, using Base64 [[RFC4648](https://datatracker.ietf.org/doc/html/rfc4648)].

It told you that you should encode the data for maximize compatibility, the example is using `base64` but it is up to the implementer.

Reply:
Again, the issue is not about the usage of decodeURI, it should of course do it when possible

Problem here is the change here that removes values that are not decodable by decodeURI, this wasn’t the case before and it invalidates valid cookie values

I believe this will break some edge use cases, for instance you might have a client that URI decodes the set values when it can, but does not URI encode it when it sends it back